### ISSUE TITLE
Property access using array notation can be quoted using ' but not "

### ISSUE THREAD

**Issue by jamie-pate** [JSONPath-Plus/JSONPath#155]
## Describe the bug

When accessing properties using array notation, you can use `'` but not `"` to enclose the property name. This would be fine, except  that `"` can be used within the query expression and it makes it confusing.

## Code sample or steps to reproduce

https://stackblitz.com/edit/js-rq1fmy?file=index.js

```js
// Code that reproduces problem here
const simple = { property: 'value' };
const arr = [simple];
// WORKS
queryAll(simple, `$['property']`);
queryAll(arr, `$[?(@['property'] == 'value')]['property']`);
queryAll(arr, `$[?(@["property"] == 'value')]['property']`);
// FAILS
queryAll(simple, `$["property"]`);
queryAll(arr, `$[?(@["property"] == "value")]["property"]`);
queryAll(arr, `$[?(@['property'] == 'value')]["property"]`);

function queryAll(json, path) {
  const result = JSONPath({ json, path });
  console.log(
    `
    ${result.length ? 'PASS' : 'FAIL'}
    expression=${path}
    array=${JSON.stringify(JSONPath.toPathArray(path))}
    json=${JSON.stringify(json)}
    result=${JSON.stringify(result)}
    `
  );
}
```

## Expected behavior

as in JavaScript, `"` should be usable to quote property names. E.G: `{ property: 'value' }` + `$["property"]` > `["value"]`.  This would make it consistent between the path expression and query expression.

## Environment (IMPORTANT)
- JSONPath-Plus version: 5.0.7

## Desktop
 - OS: Linux
 - Browser and version Chromium 91.0.4472.114 or Node Version 12.13.1

**Comment by jamie-pate** [comment]
Additionally,
`$[property]` works in the path, but `[?(@[property] == 'value')]` does not?

**Comment by brettz9** [comment]
This should now be fixed in 5.1.0 so closing. Feel free to comment further. For your other comment, feel free to file a separate issue, but as this project is not too well-maintained and is a bit hard to navigate given its frequent use of regexes rather than easily documented parsing routines, I tend to only grab the low-hanging fruit such as this issue.

**Comment by jamie-pate** [comment]
Even for issues that don't get fixed it's often helpful to have it noted since users can they confirm the issue that are facing has been encountered before and share workarounds etc :)

**Comment by jamie-pate** [comment]
I guess the follow up is: what is the npm release schedule? Seems like 5.0.7 is the latest on npm?

.. no.. wait...  5.1.0 • Public • Published 4 hours ago 🥴

**Comment by brettz9** [comment]
Yes, after your fix in 5.0.7, we added the enhancement for supporting double-quoted bracket notation. If you were pegging to the patch version 5.0.7 should work, but if you want the latest and hopefully greatest, 5.1.0 is available.
